cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -243,26 +241,10 @@ public void getStatus(HttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, status);
   }
 
-  private void getScheduleStatus(HttpResponder responder, String appId, String namespaceId, String scheduleName)
-    throws NotFoundException, SchedulerException {
-    Id.Application applicationId = Id.Application.from(namespaceId, appId);
-    ApplicationSpecification appSpec = store.getApplication(applicationId);
-    if (appSpec == null) {
-      throw new NotFoundException(applicationId);
-    }
-
-    ScheduleSpecification scheduleSpec = appSpec.getSchedules().get(scheduleName);
-    if (scheduleSpec == null) {
-      throw new NotFoundException(scheduleName, String.format("Schedule: %s for application: %s",
-                                                              scheduleName, applicationId.getId()));
-    }
-
-    String programName = scheduleSpec.getProgram().getProgramName();
-    ProgramType programType = ProgramType.valueOfSchedulableType(scheduleSpec.getProgram().getProgramType());
-    Id.Program programId = Id.Program.from(namespaceId, appId, programType, programName);
+  private void getScheduleStatus(HttpResponder responder, String namespaceId, String appId, String scheduleName)
```

Comment:
Yes but then creating json stuff has to be moved there too which seems a bit unclean so kept the method. Let me know if you are strongly against it then I will change.